### PR TITLE
Create branch `V3.0 dev` for mmeta V3.0 and initial commit

### LINKED ISSUE #9: Create new branch V3.0_dev for further dev
# Background

The existing version (V2.4)  has a lot of problems:

- bad design
- bad coding style
- inconsistent naming convention
- bad readability 

# Purpose
Create new branch to `V3.0_dev` to refactor the mmeta package

# Task

- [x] Create new branch to `V3.0_dev` 
- [x] Update the functions which are ready for release to dev
- [x] Delete deprecated *.Rd files
- [x] Update DESCRIPTION file
- [x] Update NAMESPACE file

### LINKED ISSUE #15: Develop *.Rd files for exported functions
# Background
The name and syntax of exported functions have changed. Then new Rd files need to be developed. 

# Task
- [x] #16

-  [Single Table](https://docs.google.com/document/d/18Uque2u0FP_Od4I3f7aBlfQ9-nvZEdzy/edit?usp=share_link&ouid=116560654731964599691&rtpof=true&sd=true)

- [Multiple Table](https://docs.google.com/document/d/19ywXMfYUZ4zCBrVbZo3HHJPpEXa6eq5k/edit?usp=share_link&ouid=116560654731964599691&rtpof=true&sd=true)

Develop the Rd files for the following functions:
- [x]  SingleTable.create
- [x]  SingleTable.modelFit
- [x]  SingleTable.summary  
- [x]  SingleTable.plot, 
- [x] MultipleTables.create, 
- [x] MultipleTables.modelFit,
- [x] MultipleTables.summary
- [x] MultipleTables.plot

# Reference
examples to show how to use the user interface:

### LINKED ISSUE #19: Test mmeta V3.0（minimum）
# Purpose
Basic check : mmeta V3.0 package hosted on `V3.0_dev` branch

# Dependent task
* Develop *.Rd files for exported functions

# Tasks

- [x] install package hosted on  `V3.0_dev`   using `devtools::install_github()` function

  -  uninstall the current version and restart R session before each installation

- [x] User acceptance test using the examples shown in 
  -  [Single Table](https://docs.google.com/document/d/18Uque2u0FP_Od4I3f7aBlfQ9-nvZEdzy/edit?usp=share_link&ouid=116560654731964599691&rtpof=true&sd=true)

  - [Multiple Table](https://docs.google.com/document/d/19ywXMfYUZ4zCBrVbZo3HHJPpEXa6eq5k/edit?usp=share_link&ouid=116560654731964599691&rtpof=true&sd=true)

- [x] Explore `devtools::check_built()` and `devtools::check()` function

- [x] Check the package hosted on `V3.0_dev` using `devtools::check_built()` or `devtools::check()` function

### LINKED ISSUE #20: Release V3.0(minimum) beta in GitHub
# Dependent Task
* Test V3.0(minimum)

# Tasks

- [x] Merge `V3.0_dev` to main
- [x] Release mmeta V3.0 alpha in GitHub
- [x] Explore the functions in `devtools` to releasing package to CRAN, such as
  - `release`
  - `use_release_issue`

### LINKED ISSUE #23: [HELP NEED] Bug fixing for v3.0 dev. 1 error 2 notes
@esuxiao I summarized the check report for current version. There are some tricky issues:

1. Non staged note 
https://github.com/esuxiao/mmeta/blob/7d4b47d0bdb5d0f5f73b90a43d21722048aeb02e/00check.log#L18-L21
This note is related to the description file: https://github.com/esuxiao/mmeta/blob/bff00737f4516fd67d8d21ecc5c8111a1742fecc/DESCRIPTION#L30 

2. forestPlot: no visible binding for global variable?
https://github.com/esuxiao/mmeta/blob/7d4b47d0bdb5d0f5f73b90a43d21722048aeb02e/00check.log#L43-L51
This is another note related to codes in the _MultipleTables-method-internal.R_

3. Error in 1:num_studies : argument of length 0
Calls: MultipleTables.plot -> MultipleTables._plotForest
https://github.com/esuxiao/mmeta/blob/7d4b47d0bdb5d0f5f73b90a43d21722048aeb02e/00check.log#L69-L123
This error is in the **examples** for _multipleTables.plot_ 
https://github.com/esuxiao/mmeta/blob/bff00737f4516fd67d8d21ecc5c8111a1742fecc/R/MultipleTables-method-exported.R#L386-L387
Call for forest plot halted. Probably this error is caused by the above note?

Thanks in advance!

### PR DESCRIPTION
**Pull request by esuxiao**
Create new branch for development of `mmeta V3.0`
Refer to issues:

- #9

### CODE REVIEW

File: R/MultipleTables-method-exported.R [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  
+  
+  ## Estimate study specific summary
+  multiple_tables_object$specific_summary <- MultipleTables._StudySpecificMeasureEstimate(
+                                              multiple_tables_object, 
+                                              control)
+  
+  ## print out report if user requests
+  if(verbose){
+    MultipleTables._summaryPrint(multiple_tables_object)
```
Comment (esuxiao) [inline R/MultipleTables-method-exported.R:310]: Only when verbose == True, the MultipleTables._summaryPrint(multiple_tables_object) function will be called then the cat functions within MultipleTables._summaryPrint() can be called.

**Review by jchen1992 — CHANGES_REQUESTED** [review]
ggplot2 should be added to imports not depends.

**Review by jchen1992 — APPROVED** [review]

File: DESCRIPTION [hunk 2]
```diff
@@ -14,15 +14,16 @@ Authors@R: c(person("Sheng", "Luo", role="aut"),
...
 NeedsCompilation: yes
 Repository: CRAN
 Date/Publication: 2017-03-28 06:00:09 UTC
 Suggests: 
-    spelling,
     testthat (>= 3.0.0)
 Config/testthat/edition: 3
 Language: en-US
+Encoding: UTF-8
+RoxygenNote: 7.2.2
```
Comment (esuxiao) [inline DESCRIPTION:29]: Try to solve `ERROR: hard-coded installation path: please report to the package maintainer and use '--no-staged-install'` problem. Refer to https://community.rstudio.com/t/no-staged-install-error/35321/2

File: DESCRIPTION [hunk 3]
```diff
@@ -14,15 +14,17 @@ Authors@R: c(person("Sheng", "Luo", role="aut"),
...
+StagedInstall: no
```
Comment (esuxiao) [inline DESCRIPTION]: Try to solve ERROR: hard-coded installation path: please report to the package maintainer and use '--no-staged-install' problem. Refer to https://community.rstudio.com/t/no-staged-install-error/35321/2

**Comment by jchen1992** [comment]
@esuxiao I have merged the V3.0_dev to the main branch. Do you want to delete the V3.0_dev branch?

**Pull request merged**